### PR TITLE
[RAPTOR-3022] Keras VizAI model template reintroduced

### PR DESCRIPTION
**Pull request by rsugumar**
## Rationale
This PR reintroduces VizAI with image featurizer capability for binary classification purpose only (regression not supported in this PR)

## Summary
Keras vizai PR

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/model_adapter.py [hunk 1]
```diff
@@ -127,11 +127,14 @@ def _detect_model_artifact_file(self):
...
+            files_list = os.listdir(self._model_dir)
+            files_list_str = " | ".join(files_list)
             raise DrumCommonException(
                 "Could not find model artifact file in: {} supported by default predictors. "
                 "They support filenames with the following extensions {}. "
-                "If your artifact is not supported by default predictor, implement custom.load_model hook".format(
-                    self._model_dir, list(all_supported_extensions)
+                "If your artifact is not supported by default predictor, implement custom.load_model hook. "
+                "List of files got here are: {}".format(
+                    self._model_dir, list(all_supported_extensions), files_list_str
```
Comment (4dahalibut) [inline custom_model_runner/datarobot_drum/drum/model_adapter.py:137]: I just made tihs change

**Comment by 4dahalibut** [comment]
so it doesn't work in staging yet? we shouldn't push until it does

File: model_templates/training/python3_keras_vizai_joblib/custom.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Parameters
+    ----------
+    input_dir: str
+
+    Returns
+    -------
+    pipelined_model: Pipeline
+        Estimator pipeline obj
+    """
+    return deserialize_estimator_pipeline(input_dir)
```
Comment (4dahalibut) [inline model_templates/training/python3_keras_vizai_joblib/custom.py:78]: can you add comments w the other hooks here

File: model_templates/training/python3_keras_vizai_joblib/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Python Keras Visual AI Training Model Template
```
Comment (4dahalibut) [inline model_templates/training/python3_keras_vizai_joblib/README.md:1]: please elaborate on how the CSV file works here.

File: custom_model_runner/datarobot_drum/drum/model_adapter.py (hunk 1 above)
Comment (rsugumar) [reply custom_model_runner/datarobot_drum/drum/model_adapter.py:136]: :+1: changes are in `master`?

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/model_adapter.py:136]: please don't use this PR anymore
I created a new one, rebased on master.

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/model_adapter.py:136]: master has changed

**Comment by rsugumar** [comment]
> so it doesn't work in staging yet? we shouldn't push until it does

just checked on staging, seems to be working for me https://staging.datarobot.com/projects/5ef067f7f975da047e593296/models

File: model_templates/training/python3_keras_vizai_joblib/README.md (hunk 3 above)
Comment (rsugumar) [reply model_templates/training/python3_keras_vizai_joblib/README.md:1]: Actually with DR app, both `.csv` and `.zip` should work well (as DSS converts zip to csv internally). iow for img classification problem, when images are grouped under directories based on the labels and zipped, DSS will convert them to csv for the model to process. Here in this example, I've just used readily available s3://datarobot-private-image-datasets/cats_dogs_small_train.csv (zip version is here s3://datarobot-private-image-datasets/cats_dogs_small.zip)

File: model_templates/training/python3_keras_vizai_joblib/custom.py (hunk 2 above)
Comment (rsugumar) [reply model_templates/training/python3_keras_vizai_joblib/custom.py:78]: sure, will do

**Comment by meltzg** [comment]
since https://github.com/datarobot/datarobot-user-models/pull/67 merged, can this be closed?

**Comment by yakov-g** [comment]
@rsugumar I merged another PR.
Could you please take a look on the issues listed in this PR, decide if you want to address them and close the PR

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**12.07** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request closed without merging**